public/apps/login/login-page.tsx

```diff
@@ -89,60 +98,166 @@ export function LoginPage(props: LoginPageDeps) {
     } catch (error) {
       console.log(error);
       setloginFailed(true);
+      setloginError('Invalid username or password. Please try again.');
       return;
     }
   };
 
+  const formOptions = (options: string | string[]) => {
+    let formBody = [];
+    const formBodyOp = [];
+    let authOpts = [];
+
+    if (typeof options === 'string') {
+      if (options === '') {
+        authOpts.push(AuthType.BASIC);
+      } else {
+        authOpts.push(options);
+      }
+    } else {
+      if (options && options.length === 1 && options[0] === '') {
+        authOpts.push(AuthType.BASIC);
+      } else {
+        authOpts = [...options];
+      }
+    }
+
+    for (let i = 0; i < authOpts.length; i++) {
+      switch (authOpts[i].toLowerCase()) {
+        case AuthType.BASIC: {
```

Comment:
Moved from an overall comment:

> I think you found some problems with the previous conventions where switch/casing is used instead of something like an 'authentication handler' interface. This is causing lots of potential gaps where functionality could be missing / broken and its hard to know if it is being exercised. Lets look at a couple of this places with different behavior patterns and see if we can generalize. Then a single auth handler or multiple will look identical at the different code touchpoints and it will be trivial to add more auth types in the future.

I might have missed comments in the larger conversation, making sure to capture this as a comment for dialog, what do you think of this proposal?

Reply:
Addressed.